gobblin-modules/gobblin-kafka-08/src/main/java/gobblin/kafka/writer/Kafka08DataWriter.java

```diff
@@ -74,30 +78,33 @@ public long bytesWritten() {
         }
       };
 
-  private final Producer<String, D> producer;
+  private final Producer<String, byte[]> producer;
   private final String topic;
+  private final List<StreamEncoder> encoders;
+  private final Serializer<D> valueSerializer;
 
-  public static Producer getKafkaProducer(Properties props)
+  public Producer<String, byte[]> getKafkaProducer(Properties props)
   {
     Object producerObject = KafkaWriterHelper.getKafkaProducer(props);
     try
     {
-      Producer producer = (Producer) producerObject;
+      @SuppressWarnings("unchecked")
+      Producer<String, byte[]> producer = (Producer<String, byte[]>) producerObject;
+
       return producer;
     } catch (ClassCastException e) {
       log.error("Failed to instantiate Kafka producer " + producerObject.getClass().getName() + " as instance of Producer.class", e);
       throw Throwables.propagate(e);
     }
   }
 
-  public Kafka08DataWriter(Properties props) {
-    this(getKafkaProducer(props), ConfigFactory.parseProperties(props));
-  }
 
-  public Kafka08DataWriter(Producer producer, Config config)
-  {
-    this.topic = config.getString(KafkaWriterConfigurationKeys.KAFKA_TOPIC);
-    this.producer = producer;
+  @SuppressWarnings("unchecked")
+  public Kafka08DataWriter(Properties props, List<StreamEncoder> encoders) {
+    this.producer = getKafkaProducer(props);
+    this.valueSerializer = (Serializer<D>)KafkaWriterHelper.getValueSerializer(Serializer.class, props);
```

Comment:
Please check with @shirshanka, I believe he wanted to get rid of explicit serializers.